src/frontend/components/property-type/reference/edit.tsx

```diff
@@ -13,67 +15,54 @@ type SelectRecordEnhanced = SelectRecord & {
   record: RecordJSON;
 }
 
-class Edit extends React.Component<CombinedProps> {
-  constructor(props: CombinedProps) {
-    super(props)
-    this.loadOptions = this.loadOptions.bind(this)
-    this.handleChange = this.handleChange.bind(this)
-  }
-
-  handleChange(selected: SelectRecordEnhanced): void {
-    const { onChange, property } = this.props
+const Edit: FC<CombinedProps> = (props) => {
+  const { onChange, property, record, theme } = props
+  const handleChange = (selected: SelectRecordEnhanced): void => {
     if (selected) {
       onChange(property.name, selected.value, selected.record)
     } else {
       onChange(property.name, '')
     }
   }
 
-  async loadOptions(inputValue: string): Promise<Array<SelectRecordEnhanced>> {
-    const { property } = this.props
-    const api = new ApiClient()
+  const [options, setOptions] = useState<SelectRecordEnhanced[] | null>(null)
+  useEffect(() => {
+    loadOptions(property.reference!).then(setOptions)
```

Comment:
options should be loaded also when user writes something in the input

Reply:
Alright, so this requires a completely different approach. I'll revert these changes and start from scratch, then force push on this branch.